src/bridges/convex/ConvexStakingBridge.sol

```diff
@@ -0,0 +1,265 @@
+        RCT_IMPLEMENTATION = address(new RepresentingConvexToken());
+    }
+
+    /**
+    @notice Function so the bridge can receive ether. Used for subsidy.
+    */
+    receive() external payable {}
+
+    /**
+    @notice Stake and unstake Curve LP tokens through Convex Finance Booster anytime.
+    @notice Convert rate between Curve LP token and corresponding Convex LP token is 1:1.
+    @notice Stake == Deposit, Unstake == Withdraw
+    @notice RCT (Representing Convex Token) is a representation of Convex LP token minted for bridge but fully owned by the bridge
+    @param _inputAssetA Curve LP token (staking), RCT (unstaking)
+    @param _outputAssetA RCT (staking), Curve LP token (unstaking)
+    @param _totalInputValue Total number of Curve LP tokens to deposit / withdraw
+    @param outputValueA Number of Curve LP tokens staked / unstaked, Number of RCT minted / burned
+    @param _rollupBeneficiary Address of the contract that receives subsidy
+    */
+    function convert(
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata _inputAssetB,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata _outputAssetB,
+        uint256 _totalInputValue,
+        uint256,
+        uint64,
+        address _rollupBeneficiary
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 outputValueA,
+            uint256,
+            bool
+        )
+    {
+        if (
+            _inputAssetA.assetType != AztecTypes.AztecAssetType.ERC20 ||
+            _outputAssetA.assetType != AztecTypes.AztecAssetType.ERC20
+        ) {
+            revert InvalidAssetType();
+        }
+
+        address inputDeployedClone = deployedClones[_inputAssetA.erc20Address];
+        address outputDeployedClone = deployedClones[_outputAssetA.erc20Address];
+
+        if (inputDeployedClone == address(0) && outputDeployedClone == address(0)) {
+            revert UnknownAssetA(); // invalid address or pool has not been loaded yet / RCT token not deployed yet
+        }
+
+        PoolInfo memory selectedPool;
+
+        if (inputDeployedClone == _outputAssetA.erc20Address) {
+            // deposit
+            selectedPool = pools[_inputAssetA.erc20Address];
+
+            outputValueA = _deposit(_outputAssetA, _totalInputValue, selectedPool);
+        } else if (outputDeployedClone == _inputAssetA.erc20Address) {
+            // withdrawal
+            selectedPool = pools[_outputAssetA.erc20Address];
+
+            outputValueA = _withdraw(_inputAssetA, _outputAssetA, _totalInputValue, selectedPool);
+        } else {
+            revert ErrorLib.InvalidOutputA();
+        }
+
+        _claimSubsidy(SUBSIDY, _inputAssetA, _inputAssetB, _outputAssetA, _outputAssetB, _rollupBeneficiary);
+    }
+
+    /**
+    @notice Internal function to pay out subsidy to the rollupBeneficiary
+    */
+    function _claimSubsidy(
```

Comment:
Convert complained about too many local variables - stack too deep so I decided to move this one out.

Reply:
Ok, you can solve that by moving the call up in the function.

Reply:
By removing inputDeployedClone and outputDeployedClone, I could place the subsidy claim function back inside the convert function.

Reply:
I am not sure I follow how moving the call up in the function would affect the total number of local variables?🤔 Call you please give me a more detailed explanation?

Reply:
It would affect the number of variables on the stack at the time of execution of the code on that particular line. The limitation here is that EVM can only access variables on the stack up to 16 levels deep. Vyper language doesn't have this error since they instead allocate everything to memory instead of using stack. See [this article](https://medium.com/coinmonks/stack-too-deep-error-in-solidity-608d1bd6a1ea)